### PR TITLE
Migrate to Typescript

### PR DESCRIPTION
**Pull request by hawkrives**
It's finally time to migrate this project to Typescript. While it's been nice not having any build step, I've come to yearn for the feeling of safety that I get from having `tsc` close at hand. Anyway!

- Migrate all files to `.ts` extensions
- Add types, where appropriate
- Migrate test runner to `ava` (has auto-TS stuff built in)
- Add [`typed-query-selector`](https://github.com/g-plane/typed-query-selector) for some basic querySelector checking (even works automatically with JSDOM!)
- Added a few more Zod schemæ
- Expand `prettier.printWidth` to 100 – typescript likes a little more horizontal freedom
- Perform some code cleanups

### CODE REVIEW

File: source/student-orgs/presence.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+export function cleanOrg(org: DetailedPresenceOrgType, sortableRegex: RegExp) {
+	let name = org.name.trim()
+	let category = org.categories.join(', ')
+	let meetings =
+		(org.regularMeetingLocation || '').trim() +
+		(org.regularMeetingTime || '').trim()
+	let description = JSDOM.fragment(org.description).textContent?.trim() ?? ''
+	let website = org.website?.trim() ?? ''
+	if (website && !/^https?:\/\//.test(website)) {
+		website = `http://${website}`
```
Comment (drewvolz) [inline source/student-orgs/presence.ts:42]: Why did we opt for http again?

File: source/ccc-lib/http.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import ky, {type AfterResponseHook, type BeforeRequestHook} from 'ky'
+import memoize from 'memoize'
+import {ONE_MINUTE} from './constants.js'
+
+export const USER_AGENT = 'ccc-server/0.2.0'
+
+const IS_DEBUG_KY = process.env['TRACE']?.split(',').includes('ky')
```
Comment (drewvolz) [inline source/ccc-lib/http.ts:7]: 🎉

**Review by drewvolz — APPROVED** [review]

File: source/student-orgs/presence.ts (hunk 1 above)
Comment (hawkrives) [reply source/student-orgs/presence.ts:42]: At the time, I believe there were several sites that only supported HTTP. I think we could probably do HTTPS now, without breaking anyone…

**Review by drewvolz — DISMISSED** [review]

**Review by drewvolz — APPROVED** [review]

**Review by drewvolz — DISMISSED** [review]
Nicely done @hawkrives! 🥳

**Review by drewvolz — APPROVED** [review]

**Pull request merged**